### PR TITLE
[llvm] [aot] Add numerical unit tests for LLVM-CGraph

### PR DESCRIPTION
**Pull request by jim19930609**
Related issue = #4800

### CODE REVIEW

File: tests/cpp/aot/llvm/utils.py [hunk 1]
```diff
@@ -121,3 +121,48 @@ def run(base: int, arr: ti.types.ndarray()):
...
+    def run2(base: int, arr: ti.any_arr(field_dim=1)):
+        for i in arr:
+            arr[i] += [base + i]
+
+    arr = ti.graph.Arg(ti.graph.ArgKind.NDARRAY,
+                       'arr',
+                       ti.i32,
+                       element_shape=(1, ))
+
+    base0 = ti.graph.Arg(ti.graph.ArgKind.SCALAR, 'base0', ti.i32)
```
Comment (ailzhang) [inline tests/cpp/aot/llvm/utils.py:149]: FYI the latest master will require specifying `field_dim` here as well. Fixing this should be good enough to pass all tests after rebasing on top of master.

**Review by ailzhang — APPROVED** [review]
LGTM!

**Comment by ailzhang** [comment]
FYI please don't merge this until you rebase on top of master

File: python/taichi/aot/utils.py [hunk 2]
```diff
@@ -2,11 +2,24 @@
...
+from taichi.types import f32, i32
 from taichi.types.annotations import template
 from taichi.types.ndarray_type import NdarrayType
 
 template_types = (NdarrayType, template)
 
+_aliased_type = {i32: int, int: i32, f32: float, float: f32}
+
+
+def check_type_match(lhs, rhs):
```
Comment (ailzhang) [inline python/taichi/aot/utils.py:14]: This is not accurate - try searching for `cook_dtype` (sorry naming is not that helpful :P

Comment (jim19930609) [reply python/taichi/aot/utils.py:14]: Thanks!

**Review by ailzhang — APPROVED** [review]
Thanks!!

**Pull request merged**